test/Microsoft.ML.Functional.Tests/Training.cs

```diff
@@ -0,0 +1,530 @@
+            // Create a model training an OVA trainer with a binary classifier.
+            var anomalyDetectionTrainer = mlContext.AnomalyDetection.Trainers.AnalyzeRandomizedPrincipalComponents();
+            var anomalyDetectionPipeline = mlContext.Transforms.Concatenate("Features", Iris.Features)
+                .AppendCacheCheckpoint(mlContext)
+                .Append(mlContext.Transforms.Conversion.MapValueToKey("Label"))
+                .Append(mlContext.MulticlassClassification.Trainers.OneVersusAll(anomalyDetectionTrainer))
+                .Append(mlContext.Transforms.Conversion.MapKeyToValue("PredictedLabel"));
+
+            // Fit the binary classification pipeline.
+            Assert.Throws<InvalidOperationException>(() => anomalyDetectionPipeline.Fit(data));
+
+            // Create a model training an OVA trainer with a binary classifier.
+            var binaryclassificationTrainer = mlContext.BinaryClassification.Trainers.LogisticRegression(
+                new LogisticRegression.Options { MaximumNumberOfIterations = 10, NumberOfThreads = 1, });
+            var binaryClassificationPipeline = mlContext.Transforms.Concatenate("Features", Iris.Features)
+                .AppendCacheCheckpoint(mlContext)
+                .Append(mlContext.Transforms.Conversion.MapValueToKey("Label"))
+                .Append(mlContext.MulticlassClassification.Trainers.OneVersusAll(binaryclassificationTrainer));
+
+            // Fit the binary classification pipeline.
+            var binaryClassificationModel = binaryClassificationPipeline.Fit(data);
+
+            // Transform the data
+            var binaryClassificationPredictions = binaryClassificationModel.Transform(data);
+
+            // Evaluate the model.
+            var binaryClassificationMetrics = mlContext.MulticlassClassification.Evaluate(binaryClassificationPredictions);
+
+            // Create a model training an OVA trainer with a clustering trainer.
+            var kmeansTrainer = mlContext.Clustering.Trainers.KMeans(
+                new KMeansPlusPlusTrainer.Options { MaximumNumberOfIterations = 10, NumberOfThreads = 1, });
+
+            Assert.Throws<ArgumentOutOfRangeException>(() =>
+                mlContext.Transforms.Concatenate("Features", Iris.Features)
+                    .AppendCacheCheckpoint(mlContext)
+                    .Append(mlContext.Transforms.Conversion.MapValueToKey("Label"))
+                    .Append(mlContext.MulticlassClassification.Trainers.OneVersusAll(kmeansTrainer))
+                    .Append(mlContext.Transforms.Conversion.MapKeyToValue("PredictedLabel")));
+
+            // Create a model training an OVA trainer with a multiclass classification trainer.
+            var multiclassTrainer = mlContext.MulticlassClassification.Trainers.LogisticRegression(
+                new MulticlassLogisticRegression.Options { MaximumNumberOfIterations = 10, NumberOfThreads = 1, });
+            Assert.Throws<ArgumentOutOfRangeException>(() => 
```

Comment:
>ArgumentOutOfRangeException [](start = 26, length = 27)

So in the issue you mention that this should lead to a compile time error, ideally. Are we considering doing that?

Reply:
A compile-time error is in our original spec. I'm not sure if we are planning on it, though. Let's discuss it in the issue and decide one way or another.

---
In reply to: [265232387](https://github.com/dotnet/machinelearning/pull/2921#discussion_r265232387) [](ancestors = 265232387)

Reply:
Which issue?

Reply:
#2920

---
In reply to: [265307231](https://github.com/dotnet/machinelearning/pull/2921#discussion_r265307231) [](ancestors = 265307231)